tests/treebuilders/map.cpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * MRCPP, a numerical library based on multiresolution analysis and
+ * the multiwavelet basis which provide low-scaling algorithms as well as
+ * rigorous error control in numerical computations.
+ * Copyright (C) 2020 Stig Rune Jensen, Jonas Juselius, Luca Frediani and contributors.
+ *
+ * This file is part of MRCPP.
+ *
+ * MRCPP is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MRCPP is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with MRCPP.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * For information on the complete list of contributors to MRCPP, see:
+ * <https://mrcpp.readthedocs.io/>
+ */
+
+#include "catch.hpp"
+
+#include "factory_functions.h"
+
+#include "functions/GaussPoly.h"
+#include "treebuilders/WaveletAdaptor.h"
+#include "treebuilders/grid.h"
+#include "treebuilders/map.h"
+#include "treebuilders/project.h"
+
+using namespace mrcpp;
+
+namespace mapping {
+
+template <int D> void testMapping();
+
+SCENARIO("Map a MW tree", "[map], [tree_builder]") {
+    GIVEN("One MW functions in 1D") { testMapping<1>(); }
+    GIVEN("One MW functions in 2D") { testMapping<2>(); }
+    GIVEN("One MW functions in 3D") { testMapping<3>(); }
+}
+
+template <int D> void testMapping() {
+    const double prec = 1.0e-4;
+    double alpha = 1.0;
+    double beta = 50.0;
+    double beta_ref = 100.0;
+
+    double pos_c[3] = {-0.25, 0.35, 1.05};
+    auto pos = details::convert_to_std_array<double, D>(pos_c);
+
+    GaussFunc<D> inp_func(beta, alpha, pos);
+    GaussFunc<D> ref_func(beta_ref, alpha, pos); // LUCA: weird segfault if ths becomes a GaussPoly
```

Comment:
Do we need to address this comment?

Reply:
It is strictly speaking not connected to this PR. I copied the code from the multiplication test and while changing I forgot to change the `ref_func` from `GaussPoly` to `GaussFunc` and got a segfault. I just thought it was wierd. That's why I left the cmment there.

Reply:
I'm not able to reproduce this. I changed this line to
```
GaussPoly<3> ref_func = inp_func * inp_func;
```
but got no errors or segfaults, and it passed through valgrind.